### ISSUE TITLE
podman system reset --force destroy machine

### ISSUE THREAD

**Issue by mikewwwperry** [containers/podman#15858]
https://github.com/containers/podman/blob/541655f31a5301ea042d04782d1635ad51e9f434/pkg/machine/qemu/machine.go#L1686

Podman completely broke this system. I was following along with some guides in this projects documents. When it said to start over with a reset --force, I did so. Afterwards the entire system was broken. I'm pretty certain one of the calls in the above machine.go essentially did an rm -rf /.

```
k8s@cg4:/etc/containers$ ls 
registries.conf.d 
k8s@cg4:/etc/containers$ ls -ltr 
total 1 
drwxr-xr-x 2 root root 2 Sep 18 19:39 registries.conf.d 
k8s@cg4:/etc/containers$ cd 
k8s@cg4:~$ ls 
k8s@cg4:~$ cd /etc/containers/ 
k8s@cg4:/etc/containers$ ls 
registries.conf.d 
k8s@cg4:/etc/containers$ mkdir networks 
mkdir: cannot create directory ‚Äònetworks‚Äô: Permission denied 
k8s@cg4:/etc/containers$ sudo !! 
sudo mkdir networks 
k8s@cg4:/etc/containers$ sudo podman network inspect podman | jq .[] > /etc/containers/networks/podman.json 
-bash: /etc/containers/networks/podman.json: Permission denied 
k8s@cg4:/etc/containers$ sudo podman network inspect podman | jq .[] | sudo tee /etc/containers/networks/podman.json 
{ 
  "name": "podman", 
  "id": "2f259bab93aaaaa2542ba43ef33eb990d0999ee1b9924b557b7be53c0b7a1bb9", 
  "driver": "bridge", 
  "network_interface": "cni-podman0", 
  "created": "2022-09-18T19:41:18.954497518-04:00", 
  "subnets": [ 
    { 
      "subnet": "10.88.0.0/16", 
      "gateway": "10.88.0.1" 
    } 
  ], 
  "ipv6_enabled": false, 
  "internal": false, 
  "dns_enabled": false, 
  "ipam_options": { 
    "driver": "host-local" 
  } 
} 
k8s@cg4:/etc/containers$ 
k8s@cg4:~$ cd / 
k8s@cg4:/$ sudo podman system reset --force 
ERRO[0021] unlinkat .///usr/src: device or resource busy  
ERRO[0021] unlinkat .///usr/src: device or resource busy  
-bash: /dev/null: Permission denied 
k8s@cg4:/$  
-bash: /dev/null: Permission denied 
k8s@cg4:/$  
-bash: /dev/null: Permission denied 
k8s@cg4:/$  
-bash: /dev/null: Permission denied 
k8s@cg4:/$ ls 
-bash: /bin/ls: No such file or directory 
-bash: /dev/null: Permission denied 
k8s@cg4:/$ mount 
-bash: mount: command not found 
-bash: /dev/null: Permission denied 
k8s@cg4:/$ sudo reboot 
-bash: /bin/sudo: No such file or directory 
-bash: /dev/null: Permission denied 
k8s@cg4:/$  client_loop: send disconnect: Broken pipe 
```

**Comment by Luap99** [comment]
@BitBistro-code Please provide the information requested in the issue template. What version is this? How did you install podman?

**Comment by baude** [comment]
there is no `reset`, did you mean `rm`?

**Comment by baude** [comment]
actually after re-reading, none of your commands are machine commands.  is it correct you are just running `system reset`?

**Comment by BitBistro-code** [comment]
Sorry yes it was system reset. I updated the title. It was compiled from source yesterday.

**Comment by rhatdan** [comment]
This issue here is somehow containers/storage was pointing at /?

I have no idea how that happened.
podman system reset -f 
Should be only removing graphroot.
```
$ podman info
...
  graphRoot: /home/dwalsh/.local/share/containers/storage
```
```
sudo podman info
...
  graphRoot: /var/lib/containers/storage
```

**Comment by Luap99** [comment]
Right, before attempting to fix it we should know what is causing this. I agree that DataDirPrefix does not return `/`

**Comment by BitBistro-code** [comment]
Kind of hard to say exactly, because the box was erased. I have a potential patch in [here](https://github.com/containers/podman/pull/15904), but I haven't had a good opportunity to test it yet. I also haven't fully read through the CONTRIBUTING. Next week I'll be on vacation without internet. So any progress on the patch could be delayed.

I have a couple of ideas on how it might have happened. As I was trying to get rootful podman working with a user that had sudo. The box was staged already with debian, but I had just created the user. I was cleaning up files in /etc/containers and the user may or may not have had a home directory at the time I ran. I was thinking of just turning the return error to a panic for testing, and see if I can reproduce the premise that "/" was somehow returned.
